doc/source/security/encrypting-threads.rst

```diff
@@ -0,0 +1,71 @@
+.. _encrypting-threads:
+
+================================================================================
+Working with Background Encryption Threads
+================================================================================
+
+:Availabiliity: This feature is **Experimental**.
+
+Encryption threads in the background allow you to perform some encryption and decryption tasks in real-time. 
+
+You would use encryption threads for the following purposes:
+
+* Encryption threads can encrypt existing tables in a tablespace. Encryption threads allow encryption to be applied to all or some of the existing tables, you can exclude tables from rotation, in a background process. You can encrypt existing tables with the Master key, but you must do this operation by tablespace.
+
+* Encryption threads encrypt tables with a key from a keyring. The Master key encrypts tables by a key and is stored in the encryption header of the tablespace.
+
+* Encryption threads allow key rotation. In an encryption thread rotation, the operation re-encrypts each tablespace page by page. The Master key rotation does not re-encrypt each page,  only the tablespace encryption header.
+
+If you have tablespaces encrypted with the Master key and you enable encryption threads, the tablespaces are re-encrypted with the keyring key in a background process.
+
+.. note::
+
+    After you enable encryption threads, you cannot convert the tablespaces to Master key encryption.
+
+:Availability: This feature is **Experimental** quality.
+
+.. variable:: innodb_encryption_threads
+
+    :cli: ``--innodb-encryption-threads``
+    :dyn: Yes
+    :scope: Global
+    :vartype: Numeric
+    :default: 0
+
+This variable works in combination with the :variable: `default_table_encryption` variable set to ``ONLINE_TO_KEYRING``. This variable configures the number of threads for background encryption. For the online encryption, the value must be greater than **zero**. 
+
+.. variable:: innodb_online_encryption_rotate_key_age
+
+    :cli: ``--innodb-online-encryption-rotate-key-age``
+    :dyn: Yes
+    :scope: Global
+    :vartype: Numeric
+    :default: 1
+
+By using this variable, you can re-encrypt the table encrypted using KEYRING.
+The value of this variable determines how frequently the encrypted tables should
+be re-encrypted:
+
+*  The value is **1**, the table is re-encrypted on each key rotation. 
+*  The value is **2**, the table is re-encrypted on every other key rotation.
+*  The value is **3**, the table is re-encrypted on every third key rotation.
```

Comment:
and so on ?

Reply:
added text that emphasized the integer determines the re-encryption.